File: src/de/kaffeeshare/server/plugins/LittleGamers.java
package de.kaffeeshare.server.plugins;

import java.net.URL;

import org.jsoup.nodes.Document;
import org.jsoup.select.Elements;

import de.kaffeeshare.server.exception.PluginErrorException;

public class LittleGamers extends BasePlugin {

	@Override
	public boolean match(URL url) {
		return match(url, "www.little-gamers.com");
	}

	@Override
	protected String getDescription(Document doc) {

		String description = null;

		try {
			Elements elements = doc.getElementById("comic").getAllElements();
			description  = "<img src=\"";
			description += elements.get(0).attr("src");
			description += "\" />";
		} catch (Exception e) {
			throw new PluginErrorException(this);
		}
		
		return description;
	}

	@Override
	protected String getImageUrl(Document doc) {
		return null;
	}

}

File: extract/plugins/littlegamers.go
package plugins

import (
	"strings"

	"github.com/PuerkitoBio/goquery"
	"github.com/asaskevich/govalidator"
	"github.com/koffeinsource/kaffeeshare/data"
	"golang.org/x/net/context"
	"google.golang.org/appengine/log"
)

// Littlegamers extract a comic from a littlegamers page
func Littlegamers(i *data.Item, sourceURL string, doc *goquery.Document, c context.Context) {
	if !strings.Contains(sourceURL, "www.little-gamers.com") {
		return
	}

	log.Infof(c, "Running little-gamers plugin.")

	selection := doc.Find("img#comic")

	if len(selection.Nodes) == 0 {
		log.Infof(c, "little-gamers plugin found no img#comic. "+sourceURL)
	} else {
		if len(selection.Nodes) > 1 {
			// that should actually never happen
			log.Errorf(c, "little-gamers plugin found >1 img#comic. ??? "+sourceURL)
		}
		m := htmlAttributeToMap(selection.Nodes[0].Attr)

		if govalidator.IsRequestURL(m["src"]) {
			i.Description = "<img src =\""
			i.Description += m["src"]
			i.Description += "\" />"
			i.ImageURL = ""
		} else {
			log.Errorf(c, "little-gamers plugin invalid url. "+m["src"])
		}
	}

}
